code/modules/SCP/SCPs/SCP-457.dm

```diff
@@ -0,0 +1,134 @@
+GLOBAL_LIST_EMPTY(scp457s)
+
+/datum/scp/scp_457
+	name = "SCP-457"
+	designation = "457"
+	classification = KETER
+
+/mob/living/scp_457
+	name = "SCP-457"
+	desc = "A horrible band of fire, it somehow keeps a humanoid shape."
+	icon = 'icons/SCP/scp-457.dmi'
+	icon_state = "fireguy"
+	SCP = /datum/scp/scp_457
+	status_flags = NO_ANTAG
+	var/door_cooldown
+	var/fuel = 30 //unused as time of coding
+
+	see_invisible = SEE_INVISIBLE_NOLIGHTING
+	see_in_dark = 7
+
+	maxHealth = 2000
+	health = 2000
+
+	can_pull_size = 3
+	a_intent = "harm"
+	can_be_buckled = FALSE
+
+	var/aflame_cooldown
+	var/aflame_cooldown_time = 2.8 SECONDS
+
+	var/area/spawn_area
+
+/mob/living/scp_457/Initialize()
+	GLOB.scp457s += src
+	spawn_area = get_area(src)
+	add_language(LANGUAGE_EAL, FALSE)
+	add_language(LANGUAGE_SKRELLIAN, FALSE)
+	add_language(LANGUAGE_GUTTER, FALSE)
+	add_language(LANGUAGE_SIGN, FALSE)
+	add_language(LANGUAGE_ENGLISH, TRUE)
+	set_light(0.8, 0.3, 5, l_color = COLOR_ORANGE) //makes 457 emit light
+	return ..()
+
+/mob/living/scp_457/Destroy()
+	GLOB.scp173s -= src
+	return ..()
+
+/mob/living/scp_457/UnarmedAttack(atom/A)
+	var/mob/living/carbon/human/H = A
+	if(ishuman(A))
+		if(aflame_cooldown > world.time)
+			to_chat(src, "<span class='warning'>You can't attack yet.</span>")
+			return
+		else if(aflame_cooldown < world.time)
+			if(prob(35))
+				H.Weaken(50)
+				H.visible_message("<span class='danger'>[src] claws at [H], the flame sending them to the floor!</span>")
+				to_chat(H, "<span class='userdanger'>IT HURTS!!!")
+				aflame_cooldown = world.time + aflame_cooldown_time
+				return
+			else
+				H.fire_stacks += 1
+				H.IgniteMob()
+				aflame_cooldown = world.time + aflame_cooldown_time
+				visible_message("<span class='danger'>[src] claws at [A] setting them alight!</span>")
+				to_chat(H, "<span class='userdanger'>Oh god, oh god. OH GOD! IT HURTS! PLEASE!")
+				return
+	if(H.SCP)
+		to_chat(src, "<span class='warning'><I>[H] is a fellow SCP!</I></span>")
+		return
```

Comment:
You may want to move this up in the proc above the human part, as some SCPs are human

Reply:
I am gonna assume this is intended, then? That is, being able to attack fellow SCPs that are human.

Reply:
> I am gonna assume this is intended, then? That is, being able to attack fellow SCPs that are human.

Well, no. It is not.